pydantic_ai_slim/pydantic_ai/models/openrouter.py

```diff
@@ -0,0 +1,618 @@
+        """Initialize an OpenRouter model.
+
+        Args:
+            model_name: The name of the model to use.
+            provider: The provider to use for authentication and API access. If not provided, a new provider will be created with the default settings.
+            profile: The model profile to use. Defaults to a profile picked by the provider based on the model name.
+            settings: Model-specific settings that will be used as defaults for this model.
+        """
+        super().__init__(model_name, provider=provider or OpenRouterProvider(), profile=profile, settings=settings)
+
+    @override
+    def prepare_request(
+        self,
+        model_settings: ModelSettings | None,
+        model_request_parameters: ModelRequestParameters,
+    ) -> tuple[ModelSettings | None, ModelRequestParameters]:
+        merged_settings, customized_parameters = super().prepare_request(model_settings, model_request_parameters)
+        new_settings = _openrouter_settings_to_openai_settings(cast(OpenRouterModelSettings, merged_settings or {}))
+        return new_settings, customized_parameters
+
+    @override
+    def _validate_completion(self, response: chat.ChatCompletion) -> OpenRouterChatCompletion:
+        response = OpenRouterChatCompletion.model_validate(response.model_dump())
+
+        if error := response.error:
+            raise ModelHTTPError(status_code=error.code, model_name=response.model, body=error.message)
+
+        return response
+
+    @override
+    def _process_thinking(self, message: chat.ChatCompletionMessage) -> list[ThinkingPart] | None:
+        assert isinstance(message, OpenRouterCompletionMessage)
+
+        if reasoning_details := message.reasoning_details:
+            return [_from_reasoning_detail(detail) for detail in reasoning_details]
+        else:
+            return super()._process_thinking(message)
+
+    @override
+    def _process_provider_details(self, response: chat.ChatCompletion) -> dict[str, Any]:
+        assert isinstance(response, OpenRouterChatCompletion)
+
+        provider_details = super()._process_provider_details(response)
+
+        provider_details['downstream_provider'] = response.provider
+        provider_details['native_finish_reason'] = response.choices[0].native_finish_reason
+
+        return provider_details
+
+    @override
+    def _map_model_response(self, message: ModelResponse) -> chat.ChatCompletionMessageParam:
+        texts: list[str] = []
+        tool_calls: list[chat.ChatCompletionMessageFunctionToolCallParam] = []
+        reasoning_details: list[dict[str, Any]] = []
+        for item in message.parts:
+            if isinstance(item, TextPart):
+                texts.append(item.content)
+            elif isinstance(item, ThinkingPart):
+                if item.provider_name == self.system:
+                    reasoning_details.append(_into_reasoning_detail(item).model_dump())
+                else:  # pragma: no cover
+                    pass
```

Comment:
In other model classes, we include thoughts from other providers as think tags:

https://github.com/pydantic/pydantic-ai/blob/86b645f4064202955ad297daa3cdf8dd0f777a4a/pydantic_ai_slim/pydantic_ai/models/anthropic.py#L527-L552